vulnerabilities/importers/istio.py

```diff
@@ -0,0 +1,193 @@
+        if not version_range_list:
+            return all_version, []
+        safe_pkg_versions = []
+        vuln_pkg_versions = []
+        version_ranges = [RangeSpecifier(r) for r in version_range_list]
+        for version in all_version:
+            if any([version in v for v in version_ranges]):
+                vuln_pkg_versions.append(version)
+
+        safe_pkg_versions = set(all_version) - set(vuln_pkg_versions)
+        return safe_pkg_versions, vuln_pkg_versions
+
+    def get_data_from_yaml_lines(self, yaml_lines):
+        """Return a mapping of data from a iterable of yaml_lines
+        for example :
+            ['title: ISTIO-SECURITY-2019-001',
+            'description: Incorrect access control.','cves: [CVE-2019-12243]']
+
+            would give {'title':'ISTIO-SECURITY-2019-001',
+            'description': 'Incorrect access control.',
+            'cves': '[CVE-2019-12243]'}
+        """
+
+        return yaml.safe_load("\n".join(yaml_lines))
+
+    def get_yaml_lines(self, lines):
+        """The istio advisory file contains lines similar to yaml format .
+        This function extracts those lines and return an iterable of lines
+
+        for example :
+            lines =
+            ---
+            title: ISTIO-SECURITY-2019-001
+            description: Incorrect access control.
+            cves: [CVE-2019-12243]
+            ---
+
+        get_yaml_lines(lines) would return
+        ['title: ISTIO-SECURITY-2019-001','description: Incorrect access control.'
+        ,'cves: [CVE-2019-12243]']
+        """
+
+        for index, line in enumerate(lines):
+            line = line.strip()
+            if line.startswith("---") and index == 0:
+                continue
+            elif line.endswith("---"):
+                break
+            else:
+                yield line
+
+    def process_file(self, path):
+
+        advisories = []
+
+        data = self.get_data_from_md(path)
+
+        releases = []
+        if data.get("releases"):
+            for release in data["releases"]:
+                release = release.strip()
+                release = release.split(" ")
+                if len(release) > 2:
+                    lbound = ">=" + release[0]
+                    ubound = "<=" + release[2]
+                    releases.append(lbound + "," + ubound)
+
+        data["release_ranges"] = releases
+
+        if not data.get("cves"):
+            data["cves"] = [""]
+
+        for cve_id in data["cves"]:
+
+            if not cve_id.startswith("CVE"):
+                cve_id = ""
+
+            safe_pkg_versions = []
+            vuln_pkg_versions = []
+
+            if not data.get("release_ranges"):
+                data["release_ranges"] = []
+
+            safe_pkg_versions, vuln_pkg_versions = self.get_versions_for_pkg_from_range_list(
+                                                   data["release_ranges"])
+
+            safe_purls = []
```

Comment:
Is there a need to declare this here, since you are assigning them the union below ?

Reply:
I should remove it.